src/igniteui.angular2.ts

```diff
@@ -521,9 +521,8 @@ export class IgGridBase<Model> extends IgControlBase<Model> implements AfterCont
 				this._dataSource = jQuery.extend(true, [], this._config.dataSource);
 				for (i = 0; i < diff.length; i++) {
 					for (j = 0; j < diff[i].txlog.length; j++) {
-						colIndex = element.data(this._widgetName)._getCellIndexByColumnKey(diff[i].txlog[j].key);
 						record = this._config.dataSource[diff[i].index];
-						td = element.find("tr[data-id='" + record[pkKey] + "']").children().get(colIndex);
+						td = element.data(this._widgetName)._getCellsByColKey(element.find("tr[data-id='" + record[pkKey] + "']"), diff[i].txlog[j].key);
```

Comment:
element.data(this._widgetName) is already cached in the `grid` var at the beginning. You should use it elsewhere in the function.